uaclient/cli.py

```diff
@@ -276,10 +276,17 @@ def action_detach(args, cfg):
 
     @return: 0 on success, 1 otherwise
     """
+    to_disable = []
     for ent_cls in entitlements.ENTITLEMENT_CLASSES:
         ent = ent_cls(cfg)
         if ent.can_disable(silent=True):
-            ent.disable(silent=True)
+            to_disable.append(ent)
+    if to_disable:
+        print("Detach will disable the following services:")
```

Comment:
Given the original issue I think we need to handle the optional pluralization:
plural = "s" if len(to_disable) > 1 else ""
print("Detach will disable the following service{}:".fornat(plural))

Reply:
Good catch, thanks!